### PR TITLE
docs: minimal client-server example

### PR DESCRIPTION
**Pull request by michielbdejong**
```
node test.js
server connects...
client connects...
(node:35835) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: {"code":"F00","name":"NotAcceptedError","triggeredAt":"2017-12-20T07:13:05.478Z","data":"{\"message\":\"invalid auth token and/or username\"}","protocolData":[]}
(node:35835) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
^C

### CODE REVIEW

**Comment by dappelt** [comment]
The example you added works. Why did you change the auth token code?

**Comment by michielbdejong** [comment]
> Why did you change the auth token code?

I didn't change how it works, I think? It was already looking at `options.incomingSecret` instead of `options.token` in the master branch?

My aim was to remove dead code and fix the tests to test the code path that is actually being used.

File: test/infoSpec.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ describe('Info', () => {
     this.mockSocket
       .reply(btpPacket.TYPE_MESSAGE, ({ requestId }) => btpPacket.serializeResponse(requestId, []))
 
-    await this.plugin.addSocket(this.mockSocket, { username: 'user', token: 'placeholder' })
+    await this.plugin.addSocket(this.mockSocket, { username: '', token: 'placeholder' })
```
Comment (dappelt) [inline test/infoSpec.js:36]: `username` and `token` are the same as what is defined in options.server. Better to declare a constant for username and token at the top and reference the constant.

Thinking about it some more, is there any reason `username` and `token` have to be passed to `addSocket`? The plugin could parse this information from the `options.server`, no?

**Comment by dappelt** [comment]
Removing dead auth code is definitely a good idea. @sharafian, any reason to keep `_getAuthToken`?

**Review by sharafian — APPROVED** [review]

**Comment by sharafian** [comment]
I thought getAuthToken was used at constructor time to set the auth token that the RPC class will use?

**Comment by michielbdejong** [comment]
setAuthToken was setting an unused variable `plugin._rpc._token`

**Pull request merged**